### DISCUSSION TITLE
null-coalescing collection literals

### DISCUSSION THREAD

**Discussion by RenderMichael** [dotnet/csharplang#7349] [Q&A] [answered]
There are quite a few circumstances where a method retrieves a list, but if that list is null we instead use an empty list.

I was wondering if the current status of collection literals would allow a syntax like
```csharp
object[] m1 = M() ?? []; // Would this be legal?
var m2 = M() ?? []; // Maybe even interpreting the return type of M()?

static object[]? M() { /*...*/ }
```

Sorry to rehash if this was explicitly discussed in the LDM, but that would be an extremely useful syntax for me.

**Comment by CyrusNajmabadi** [comment, marked as answer]
Please discuss things related to collection literals in the collection literals discussion :) 

To answer your question.  Both of those would be legal.
